pkg/objstore/objstore.go

```diff
@@ -22,7 +22,7 @@ type Bucket interface {
 	BucketReader
 
 	// Upload the contents of the reader as an object into the bucket.
-	Upload(ctx context.Context, name string, r io.Reader) error
+	Upload(ctx context.Context, name string, r io.Reader, sizeBytes int64) error
```

Comment:
Let's comment properly here, why this has size and reader interface. Worth to mention PR https://github.com/improbable-eng/thanos/pull/617 as we have reference discussion why etc.

Reply:
I would also add a comment saying `TODO(PR #617): Consider removing requirement on pre-known length when all providers will support this.`